core/src/main/scala/neotypes/utils.scala

```diff
@@ -1,23 +1,32 @@
 package neotypes.utils
 
 import java.util.concurrent.CompletionException
+import java.util.function.{Consumer => JConsumer, Function => JFunction}
 
 object FunctionUtils {
-  implicit def function[T, U](f: Function[T, U]): java.util.function.Function[T, U] = new java.util.function.Function[T, U] {
-    override def apply(t: T): U = f(t)
-  }
-
-  implicit def consumer[T](f: Function[T, Unit]): java.util.function.Consumer[T] = new java.util.function.Consumer[T] {
-    override def accept(t: T): Unit = f(t)
-  }
+  implicit def function[T, U](f: Function[T, U]): JFunction[T, U] =
+    new JFunction[T, U] {
+      override def apply(t: T): U = f(t)
+    }
 
+  implicit def consumer[T](f: Function[T, Unit]): JConsumer[T] =
+    new JConsumer[T] {
+      override def accept(t: T): Unit = f(t)
+    }
 }
 
 object CompletionUtils {
-  def exceptionally(fn: Throwable => Unit) = new java.util.function.Function[Throwable, Void] {
-    override def apply(e: Throwable): Void = e match {
-      case _: CompletionException => fn(e.getCause).asInstanceOf[Void]
-      case _ => fn(e).asInstanceOf[Void]
+  implicit def exceptionally(fn: Throwable => Unit): JFunction[Throwable, Void] =
+    new JFunction[Throwable, Void] {
+      override def apply(e: Throwable): Void = {
+        // Execute the function.
+        e match {
+          case _: CompletionException => fn(e.getCause)
+          case _                      => fn(e)
+        }
+        // Return null, which is the only value that conforms to Void.
+        // See: https://stackoverflow.com/questions/44171262/convert-scala-unit-to-java-void/44172467#44172467
+        None.orNull
```

Comment:
can we return `null`?

Reply:
If you want.

I am used to work with linters and fatal warnings. Thus, any explicit null will break the compile.
For those cases, I always use `None.orNull` which as the [scaladoc](https://www.scala-lang.org/api/current/scala/None$.html#orNull[A1%3E:A](implicitev:Null%3C:%3CA1):A1) says, such method was made to avoid an explicit use of null.

Reply:
Makes sense. Let's leave `None.orNull`